sync/server.md

```diff
@@ -0,0 +1,404 @@
+# Sync Server
+
+## Terminology
+
+**GraphQL** - A data query language
+
+**IDL/SDL** - Interface/Schema Definition Language. The language a GraphQL Data Schema is written in.
+
+**Client** - Any device or browser that sends a GraphQL Query or Mutation request to the server
+
+**Data Source** - A source for loading/saving data e.g. a Postgres database, a REST API, an OpenWhisk action
+
+**Resolver Mapping** - The definition for how to load/save data to a Data Source based on parameters sent by a Client, and how to format it before returning it to the client. All mapping defintions are compiled and executed as a Handlebars template. This allows for some logic when defined the request & repsonse mapping e.g. using the `toJson` help to stringify the result of an SQL query.
+
+**Resolver Request Mapping** - The defintion for how to load/save data to a Data Source based on parameters sent by a Client e.g. a SQL query, a HTTP endpoint, method & body.
+
+**Resolver Response Mapping** - The definition for how to format data retrieved from a Data Source before returning it to the client e.g. simply stringify an array of objects.
+
+**Connector** - The combination of a Data Source and Resolver Mapping capabilities e.g. a Postgres Connector would allow a developer to connect to a Postgres database, and define SQL queries to execute when particular GraphQL Queries or Mutations are called.
+
+More information about the GraphQL terminology and schema can be found at https://graphql.org/learn/schema/
+
```

Comment:
What's a "schema"?  https://github.com/aerogear/aerogear-data-sync-server/blob/0a6b9941e28fe3ff3be7d815ab96629e7befeacc/examples/schema.example.graphql

Reply:
A GraphQL Schema is the defintion/interface for all Types in your system, and the interactions that can be made on them.

This blog post is quite good at explaining https://www.prisma.io/blog/graphql-server-basics-the-schema-ac5e2950214e/

The schema you linked to is a very simple example. 
There is 1 entity/model, `Note`, with 4 fields, `id`, `title`, `content`, `timestamp`.
You can list & read Notes (Queries)
and you can create, update & delete Notes (Mutations)

GraphQL Resolvers have the actual implementation for a given Query or Mutation.

Reply:
I meant, maybe document that as well in the terminology :)